### PR TITLE
can't annotate class

### PR DESCRIPTION
**Pull request by EliosPride**

### CODE REVIEW

File: src/main/java/mate/jdbc/Main.java [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
 
+import mate.jdbc.dao.ManufacturerDaoImpl;
+import mate.jdbc.lib.Injector;
+import mate.jdbc.model.Manufacturer;
+
 public class Main {
+    private static final Injector injector = new Injector("manufacturers");
+
     public static void main(String[] args) {
+        ManufacturerDaoImpl manufacturerDao = (ManufacturerDaoImpl) injector.getInstance(ManufacturerDaoImpl.class);
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/Main.java]: @EliosPride use interface type ManufacturerDao to create instance of dao

File: src/main/java/mate/jdbc/lib/Injector.java [hunk 2]
```diff
@@ -4,17 +4,13 @@
...
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
 public class Injector {
     private static final Map<String, Injector> injectors = new HashMap<>();
     private final List<Class<?>> classes = new ArrayList<>();
 
-    private Injector(String mainPackageName) {
+    public Injector(String mainPackageName) {
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/lib/Injector.java]: @EliosPride this is utility class, it must have private constructor. It has static method, so use it like this Injector.getInstance()

File: src/main/java/mate/jdbc/Main.java [hunk 3]
```diff
@@ -1,7 +1,27 @@
 package mate.jdbc;
 
+import mate.jdbc.dao.ManufacturerDaoImpl;
+import mate.jdbc.lib.Injector;
+import mate.jdbc.model.Manufacturer;
+
 public class Main {
+    private static final Injector injector = new Injector("manufacturers");
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/Main.java]: @EliosPride dont create Injector instance, use it static methods in place where you need

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> getAllManufacturers = new ArrayList<>();
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java]: @EliosPride getAllManufacturers -> manufacturers

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        try (Connection connection = ConnectionUtil.getConnection();
+             Statement getAllInitsStatement = connection.createStatement()) {
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java]: @EliosPride getAllInitsStatement -> statement (rename in all methods)

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return getAllManufacturers;
+    }
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerRequest = "INSERT INTO manufacturers(name, country) values(?, ?);";
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java]: @EliosPride move all sql queries to constants

File: src/main/java/mate/jdbc/util/ConnectionUtil.java [hunk 7]
```diff
@@ -1,5 +1,27 @@
...
+            throw new RuntimeException("Can't load JDBC driver for MySQL", e);
+        }
+    }
 
+    public static Connection getConnection() {
+        try {
+            Properties dbProperties = new Properties();
+            dbProperties.put("user", "root");
+            dbProperties.put("password", "Aktersobaka1992");
+            return DriverManager.getConnection("jdbc:mysql://localhost:3306/init_db", dbProperties);
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/util/ConnectionUtil.java:22]: @EliosPride init_db is bad name for database schema, name it like 'taxi' or 'taxi_service' or 'taxi_app'.
You had to create init_db.sql file to run it and init your empty DB with some start data (from your task - Create init_db.sql file in src/main/resources folder.)

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
+package mate.jdbc.dao;
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java]: @EliosPride move to package 'impl'

File: src/main/java/mate/jdbc/util/DataProcessingException.java [hunk 9]
```diff
@@ -1,4 +1,4 @@
-package mate.jdbc;
+package mate.jdbc.util;
```
Comment (Garret-vin) [inline src/main/java/mate/jdbc/util/DataProcessingException.java]: @EliosPride move to package 'exception'

File: src/main/java/mate/jdbc/Main.java (hunk 1 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/Main.java]: @Garret-vin ++

File: src/main/java/mate/jdbc/lib/Injector.java (hunk 2 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/lib/Injector.java]: @Garret-vin ++

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java (hunk 4 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java]: @Garret-vin ++

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java (hunk 5 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java]: @Garret-vin ++

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java (hunk 6 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java]: @Garret-vin ++

File: src/main/java/mate/jdbc/util/DataProcessingException.java (hunk 9 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/util/DataProcessingException.java]: @Garret-vin ++

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java (hunk 8 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java]: @Garret-vin ++

**Pull request merged**